### PR TITLE
avoid mutating parent when rendering if expression

### PR DESCRIPTION
**Pull request by ginter**

### CODE REVIEW

File: sqlglot/generator.py [hunk 1]
```diff
@@ -903,7 +903,9 @@ def unique_sql(self, expression):
         return f"UNIQUE ({columns})"
 
     def if_sql(self, expression):
-        return self.case_sql(exp.Case(ifs=[expression], default=expression.args.get("false")))
+        case = exp.Case(ifs=[], default=expression.args.get("false"))
```
Comment (tobymao) [inline sqlglot/generator.py:906]: this is tricky, i see you’re doing jt because the init mutate

Comment (tobymao) [reply sqlglot/generator.py:906]: i don’t know if we should do it like this. may be cleaner to do a copy? i need to think

**Review by barakalon — COMMENTED** [review]
This is a really interesting bug.

I think we do this in other places too.

To recap from offline discussion: calling .sql() on an expression can implicitly change the parent-child relationships in the expression, since we set parents in the Expression constructor.

Is there a more elegant solution?

I.e...

```python
class Expression:
    def __init__(self, _set_parent=True, **args):
        ...
```

File: sqlglot/generator.py (hunk 1 above)
Comment (ginter) [reply sqlglot/generator.py:906]: copy is easier and also works just figured you didn't want the overhead

**Comment by tobymao** [comment]
so a copy in generator.

**Comment by barakalon** [comment]
Ah sure, that's probably better.

**Review by barakalon — APPROVED** [review]

File: sqlglot/generator.py (hunk 1 above)
Comment (tobymao) [reply sqlglot/generator.py]: i don't like overhead but, alas

**Pull request merged**